pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -132,20 +133,26 @@ func (d *DependenciesDistributor) OnAdd(obj interface{}) {
 
 // OnUpdate handles object update event and push the object to queue.
 func (d *DependenciesDistributor) OnUpdate(oldObj, newObj interface{}) {
-	unstructuredOldObj, err := helper.ToUnstructured(oldObj)
+	unstructuredNewObj, err := helper.ToUnstructured(newObj)
 	if err != nil {
-		klog.Errorf("Failed to transform oldObj, error: %v", err)
+		klog.Errorf("Failed to transform newObj, error: %v", err)
+		return
+	}
+	if !unstructuredNewObj.GetDeletionTimestamp().IsZero() {
+		klog.V(4).Infof("Ignore update event of object(%s, kind=%s, %s) as being deleted.",
+			unstructuredNewObj.GetAPIVersion(), unstructuredNewObj.GetKind(), names.NamespacedKey(unstructuredNewObj.GetNamespace(), unstructuredNewObj.GetName()))
```

Comment:
It's better to check it in the reconciling loop in case there are some clean-up tasks.

Reply:
but it's for resource template, we won't have any cleanup.

Reply:
+1

Our current model cannot discard this event, otherwise it would not be possible to do the cleanup.

Other parts LGTM.

Reply:
For this part:

https://github.com/karmada-io/karmada/blob/d676996b240f43ca0430d492b8c54eb0b4986c2d/pkg/dependenciesdistributor/dependencies_distributor.go#L256-L263

Reply:
I don't think there is any negative effect since a independent binding's delete event will handle it.

Reply:
We have added a finalizer. The delete event does not handle this.

Reply:
> I don't have a strong opinion just feel not good to do such things in event filter.
I'm not sure that's a good pattern, my thinking is the event filter should be simple.

I don't totally agree. if unnecessary events can be filtered out before they are queued, this can significantly improve the efficiency of the work queue. Of course, keeping it simple is also a consideration, but compared to this type of processing in the detector, this obviously does not add much complexity.
https://github.com/karmada-io/karmada/blob/6fbdd9e7dae2547e9255e65890abcb56b2e8a34b/pkg/detector/detector.go#L907-L953